certify-service/src/main/java/io/mosip/certify/credential/W3cJsonLd.java

```diff
@@ -0,0 +1,143 @@
+            validFrom = j.getJsonObject().get(VCDM1Constants.ISSUANCE_DATE).toString();
+        } else if (j.getJsonObject().containsKey(VCDM2Constants.VALID_FROM)){
+            validFrom = j.getJsonObject().get(VCDM2Constants.VALID_FROM).toString();
+        } else {
+            validFrom = ZonedDateTime.now(ZoneOffset.UTC)
+                    .format(DateTimeFormatter.ofPattern(Constants.UTC_DATETIME_PATTERN));
+        }
+        // TODO: VC Data Model spec doesn't specify a single date format or a
+        //  timezone restriction, this will have to be supported timely.
+        Date createDate = Date
+                .from(LocalDateTime
+                        .parse(validFrom,
+                                DateTimeFormatter.ofPattern(Constants.UTC_DATETIME_PATTERN))
+                        .atZone(ZoneId.systemDefault()).toInstant());
+
+        CertificateResponseDTO certificateResponseDTO = didDocumentUtil.getCertificateDataResponseDto(appID, refID);
+        String kid = certificateResponseDTO.getKeyId();
+        if (CertifyIssuanceServiceImpl.keyChooser.containsKey(signatureCryptoSuite)) {
+            // legacy signature algos such as Ed25519Signature{2018,2020}
+            ProofGenerator proofGenerator = proofGeneratorFactory.getProofGenerator(signatureCryptoSuite)
+                    .orElseThrow(() ->
+                            new CertifyException("Proof generator not found for algorithm: " + signatureCryptoSuite));
+            LdProof vcLdProof = LdProof.builder().defaultContexts(false).defaultTypes(false).type(proofGenerator.getName())
+                    .created(createDate).proofPurpose(VCDMConstants.ASSERTION_METHOD)
+                    .verificationMethod(URI.create(didUrl + "#" + kid))
+                    .build();
+            LdProof ldProofWithJWS = CredentialUtils.generateLdProof(vcLdProof, j,
+                    keyReferenceDetails, proofGenerator);
+            ldProofWithJWS.addToJsonLDObject(j);
+        } else {
+            LdSigner signer = LdSignerRegistry.getLdSignerByDataIntegritySuiteTerm(SignatureAlg.DATA_INTEGRITY);
+            KeymanagerByteSigner keymanagerByteSigner = KeymanagerByteSignerFactory.getInstance(appID, refID, signatureService, signAlgorithm);
+            signer.setSigner(keymanagerByteSigner);
+            signer.setCryptosuite(signatureCryptoSuite);
+
+            DataIntegrityProof dataIntegrityProof = DataIntegrityProof.builder()
+                    .created(createDate)
+                    .proofPurpose(VCDMConstants.ASSERTION_METHOD)
+                    .cryptosuite(signatureCryptoSuite)
+                    .verificationMethod(URI.create(didUrl + "#" + kid))
+                    .type(SignatureAlg.DATA_INTEGRITY).build();
+
+            dataIntegrityProof = CredentialUtils.generateDataIntegrityProof(dataIntegrityProof, j, signer);
+            dataIntegrityProof.addToJsonLDObject(j);
+        }
```

Comment:
Critical: this entire else condition on dataIntegrityProof, should this not be just another implementation of proof generator ?

Reply:
Lets create a JIRA for next release.